src/southbridge/amd/sb700/sm.c

```diff
@@ -310,6 +310,13 @@ static void sm_init(struct device *dev)
 
 		byte |= 1 << 3;
 		pci_write_config8(dev, 0x43, byte);
+
+		/* Enable southbridge MMIO decode */
+		dword = pci_read_config32(dev, SB_MMIO_CFG_REG);
+		dword &= ~(0xffffff << 8);
+		dword |= SB_MMIO_BASE_ADDRESS;
+		dword |= 0x1;
+		pci_write_config32(dev, SB_MMIO_CFG_REG, dword);
```

Comment:
Does it have to be hardcoded? Part of resource allocator v3 abuse was that there was too many hardcoded resources that were in fact configurable, like this one. Were devices under southbridge accessible during `read_resource` phase before this change?

Reply:
Actually, what is the use of setting BAR if it isn't enabled? Only IO is, MMIO enable is bit 1.

Reply:
I have changed the address to 0xfed80000 like other AMD SBs do. Also these enable bits does not seem to work as we think. At least this is what I noticed. Mayeb it needs further checks.

Reply:
Should it be ORed with 0x2 instead?

Reply:
Not sure why, this was in the original code from 4.11. I may clean it up.

Reply:
@krystian-hebel I hardcoded it to 0xfed80000 because of the and commonblock for this mmio, which can be potentially used.

Reply:
I have moved the SB MMIO init to bootblock. Later we could select the SOC_AMD_COMMON_BLOCK_ACPIMMIO and replace all the occurrences of reimplemented SB MMIO access through I/O.